tests/rptest/tests/partition_reassignments_test.py

```diff
@@ -195,8 +274,321 @@ def test_reassignments(self):
+                    all_node_idx, reassignments[topic][pid])
+                reassignments[topic][pid].append(missing_node_idx)
+
+        self.logger.debug(
+            f"Even replica count by adding. Expect fail. New assignments: {reassignments}"
+        )
+
+        def try_even_replication_factor(topics):
+            try:
+                kcl.alter_partition_reasssignments(topics)
+                raise Exception(
+                    "Even replica count accepted but it should be rejected")
+            except RuntimeError as ex:
+                if str(ex) == "Number of replicas != topic replication factor":
+                    pass
+                else:
+                    raise
+
+        try_even_replication_factor(reassignments)
+
+        # Test even replica count by removing two replicas. Expect a failure
+        # because even replication factor is not supported in Redpanda
+        for topic in reassignments:
+            for pid in reassignments[topic]:
+                reassignments[topic][pid].pop()
+                reassignments[topic][pid].pop()
+
+        self.logger.debug(
+            f"Even replica count by removal. Expect fail. New assignments: {reassignments}"
+        )
+        try_even_replication_factor(reassignments)
+
+        for prod in producers:
+            prod.wait()
+
+    @cluster(num_nodes=6)
+    def test_reassignments_cancel(self):
+        initial_assignments, all_node_idx, producers = self.initial_setup_steps(
+            producer_config={
+                "topics": [self.topics[0].name, self.topics[1].name],
+                "msg_size": 512 * 1024,
+                "msg_count": 1024
+            },
+            # Set a low throttle to slowdown partition move enough that there is
+            # something to cancel
+            recovery_rate=1000)
+
+        reassignments_json = self.make_reassignments_for_cli(
+            all_node_idx, initial_assignments)
+
+        kafka_tools = KafkaCliTools(self.redpanda)
+        output = kafka_tools.execute_reassign_partitions(
+            reassignments=reassignments_json)
+        self.logger.debug(output)
+        check_execute_reassign_partitions(output, reassignments_json,
+                                          self.logger)
+
+        output = kafka_tools.cancel_reassign_partitions(
+            reassignments=reassignments_json)
+        check_cancel_reassign_partitions(output, reassignments_json,
+                                         self.logger)
+
+        output = kafka_tools.verify_reassign_partitions(
+            reassignments=reassignments_json)
+        nodes_as_str = [str(n) for n in all_node_idx]
+        check_verify_reassign_partitions(output, reassignments_json,
+                                         nodes_as_str, self.logger)
+
+        for prod in producers:
+            prod.wait()
+
+    @cluster(num_nodes=5)
+    def test_reassignments_on_stopped_nodes(self):
```

Comment:
Instead of remaking the node_interruption tests, we could instead test that the AlterPartiton/ListPartition API is returning the correct error_code.